domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/AccessProfileServiceImpl.java

```diff
@@ -53,7 +53,7 @@ public UserRole getRole(final String role) {
             entity = toEntity(userRole);
             roleFound = false;
         }
-        return new ServiceResponse<>(toModel(repository.save(entity)), roleFound ? UPDATE : CREATE);
+        return new ServiceResponse<>(toModel(repository.saveAndFlush(entity)), roleFound ? UPDATE : CREATE);
```

Comment:
any particular reason for flush here? ( i didn't see our logic needs to read the saved changes at a later point during the same transaction)

Reply:
shouldSaveUserRole() is failing because of createdAt property. As insert is kept in hibernate memory, and createdAt is null at this point and we return the mapped object in the transaction, we are getting nullPointerException for createdAt property. As I understand from the journey, we need to return createdAt to the client, so I directly flushed it and got createdAt's value.

Reply:
Ok, i think that model mapper should be outside of the Transaction actually - existing tech debt :(